### PR TITLE
Add pass provider

### PR DESCRIPTION
**Pull request by mcanevet**
Add a new provider for `pass` (http://www.passwordstore.org/).

Works like the `Vault` provider but uses pass.

### CODE REVIEW

**Comment by mcanevet** [comment]
@stack72 is there something preventing this PR to be merged?

File: builtin/providers/pass/data_source_password_test.go [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+	"testing"
+
+	r "github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+)
+
+func TestDataSourcePassword(t *testing.T) {
+	r.Test(t, r.TestCase{
+		Providers: testProviders,
+		//PreCheck:  func() { testAccPreCheck(t) },
```
Comment (kwilczynski) [inline builtin/providers/pass/data_source_password_test.go]: Please more code that is not needed.

File: builtin/providers/pass/resource_pass_password.go [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+			"path": &schema.Schema{
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "Full path where the pass password will be written.",
+			},
+
+			// Data is passed as JSON so that an arbitrary structure is
+			// possible, rather than forcing e.g. all values to be strings.
+			"data": &schema.Schema{
```
Comment (kwilczynski) [inline builtin/providers/pass/resource_pass_password.go]: Would this benefit from JSON validation helper to be added here?

**Comment by kwilczynski** [comment]
@mcanevet would this require someone to install to get the `pass` binary and for the binary to be available in the `PATH`?

**Comment by mcanevet** [comment]
@kwilczynski yes, this provider will not work without the `pass` program available in the `PATH`.

File: builtin/providers/pass/resource_pass_password.go (hunk 2 above)
Comment (mcanevet) [inline builtin/providers/pass/resource_pass_password.go]: I'm not sur this is required, as if the password can't be unmarshaled, we fallback to plain text.

**Comment by mcanevet** [comment]
@stack72 is there something preventing this PR to be merged?

**Comment by raphink** [comment]
@stack72 is this missing something to be merged?

**Comment by mcanevet** [comment]
@kwilczynski is there something preventing this PR to be merged?

**Comment by mcanevet** [comment]
Could anyone take a look at this?
I can perfectly understand that you don't want to merge this because you don't want to maintain this provider on the long term, but if so, please tell me so that I close this PR and maintain it as a plugin instead.
/cc @stack72 @kwilczynski

**Comment by beddari** [comment]
We're another org interested in the outcome of this. Not sure about what the current policy is, however I'm pretty certain the Terraform using pass community is large enough to maintain this 👍

**Comment by mcanevet** [comment]
https://github.com/camptocamp/terraform-provider-pass

**Pull request closed without merging**